### PR TITLE
Bypass mouse scroll debounce

### PR DESCRIPTION
**Pull request by ev-elucidbio**
…ce for mouse scroll events

Exposed a function to allow the consuming system to bypass the debounce on the scroll. The bypass needs to be called explicitly, allowing backwards compatability

### Context
We were encountering an issue with the scroll events being debounced. Our back end wanted every scroll from the remote view.

### Results

This change exposes a function that will allow the consumer to bypass the debounce. It must be called explicitly. This decision was made so that it would be impossible to accidentally turn off the debounce and potentially overload the server.
### Changes

- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing

I manually tested this change. Can be verified by calling the following function from the `RemoteView` 
```
setMouseScrollDebounceBypass(true);
```
and observing the traffic on the webserver for mouse wheel events.

- [ ] This change adds or fixes unit tests 
- [x] Tested environment:
  - **vtk.js**: master
  - **OS**: Ubuntu 22.04.2 LTS
  - **Browser**: Chrome 110.0.5481.100, FireFox 110.0

### CODE REVIEW

**Comment by jourdain** [comment]
Any reason for not using the macro for the set/get?

File: Sources/Rendering/Core/RenderWindowInteractor/index.js [hunk 1]
```diff
@@ -1108,6 +1114,10 @@ function vtkRenderWindowInteractor(publicAPI, model) {
     superDelete();
   };
 
+  publicAPI.setMouseScrollDebounceBypass = (byPass) => {
+    model.bypassDebounce = byPass;
+  };
```
Comment (ev-elucidbio) [inline Sources/Rendering/Core/RenderWindowInteractor/index.js:1119]: @jourdain - if I understand what you're asking for https://github.com/Kitware/vtk-js/pull/2723#issuecomment-1441073973 correctly, you're asking if I can remove this and do something like
```
vtkRenderWindowInteractor.extend('vtkRenderWindowInteractor', 'bypassDebounce', true);
```
from the `RemoteView` function?

Comment (jourdain) [reply Sources/Rendering/Core/RenderWindowInteractor/index.js:1119]: No sorry, I meant in `Sources/Rendering/Core/RenderWindowInteractor/index.js` why you are not using the (set) macro that will trigger a `modified()` when `setMouseScrollDebounceBypass` get called. (That means you do not need to write that function. 

Then for the other classes, can we just leave it as-is and rely on the fact that you get the interactor from getters and set it on the interactor only?

Comment (jourdain) [reply Sources/Rendering/Core/RenderWindowInteractor/index.js:1119]: Basically just doing the following change (+updating the TS).

__Sources/Rendering/Core/RenderWindowInteractor/index.js__

```javascript

const DEFAULT_VALUES = {
	// ...
	mouseScrollDebounceByPass: false,
}

export function extend(publicAPI, model, initialValues = {}) {
	// ...
	macro.setget(publicAPI, model, [
            'mouseScrollDebounceByPass',
        ]);
	// ...
}

```

Comment (jourdain) [reply Sources/Rendering/Core/RenderWindowInteractor/index.js:1119]: And that part that I forgot

```js
    if (model. mouseScrollDebounceByPass) {
      publicAPI.extendAnimation(600);
      publicAPI.endMouseWheelEvent();
      model.wheelTimeoutID = 0;
    } else {
      // start a timer to keep us animating while we get wheel events
      model.wheelTimeoutID = setTimeout(() => {
        publicAPI.extendAnimation(600);
        publicAPI.endMouseWheelEvent();
        model.wheelTimeoutID = 0;
      }, 200);
    }
```

Comment (ev-elucidbio) [reply Sources/Rendering/Core/RenderWindowInteractor/index.js:1119]: will implement. Thank you!

File: Sources/Rendering/Core/RenderWindowInteractor/index.d.ts [hunk 2]
```diff
@@ -1182,9 +1184,10 @@ export interface vtkRenderWindowInteractor extends vtkObject {
 
 	/**
 	 * Allow system to bypass scrolling debounce. This function must be called to allow the debounce to be bypassed
+	 * @param mouseScrollDebounceByPass
 	 */
 
-	setMouseScrollDebounceBypass(bypass:boolean): void;
+	setMouseScrollDebounceByPass(mouseScrollDebounceByPass:boolean): void;
```
Comment (jourdain) [inline Sources/Rendering/Core/RenderWindowInteractor/index.d.ts:1190]: Need the get as well

File: Sources/Rendering/Misc/RemoteView/index.d.ts [hunk 3]
```diff
@@ -126,7 +126,7 @@ export interface vtkRemoteView extends vtkObject {
    * 
    *  @param bypass
    */
-  setMouseScrollDebounceBypass(bypass: boolean): void;
+  setMouseScrollDebounceByPass(bypass: boolean): void;
```
Comment (jourdain) [inline Sources/Rendering/Misc/RemoteView/index.d.ts:129]: Can we just skip that method and call `remoteView.getInteractor().setMouseScrollDebounceByPass(true)` instead?

File: Sources/Rendering/Misc/RemoteView/index.js [hunk 4]
```diff
@@ -93,8 +93,8 @@ function vtkRemoteView(publicAPI, model) {
     publicAPI.modified();
   };
 
-  publicAPI.setMouseScrollDebounceBypass = (bypass) => {
-    model.interactor.setMouseScrollDebounceBypass(bypass);
+  publicAPI.setMouseScrollDebounceByPass = (bypass) => {
+    model.interactor.setMouseScrollDebounceByPass(bypass);
```
Comment (jourdain) [inline Sources/Rendering/Misc/RemoteView/index.js:97]: Can we just skip that method and call `remoteView.getInteractor().setMouseScrollDebounceByPass(true)` instead?

**Comment by jourdain** [comment]
Did you test it? Does that still work fine for you?

**Comment by ev-elucidbio** [comment]
> Did you test it? Does that still work fine for you?

Yes

**Comment by ev-elucidbio** [comment]
image

image

**Comment by ev-elucidbio** [comment]
closing - opened another pr due to an issue with a commit message

**Pull request closed without merging**